src/main/java/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -35,6 +35,18 @@ public List<BookDto> findAll() {
     public BookDto findById(Long id) {
         return bookRepository.findById(id)
                 .map(bookMapper::toDto)
-                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id " + id));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto updateBookRequestDto) {
+        Book updatedBook = bookMapper.toBook(updateBookRequestDto);
+        updatedBook.setId(id);
+        return bookMapper.toDto(updatedBook);
     }
```

Comment:
You don't actually update books in your DB, just returning same book with new id.

Reply:
agree))